### ISSUE TITLE
Get Item fails on id-field for OGR Driver with GeoJSON Lakes dataset

### ISSUE THREAD

**Issue by justb4** [geopython/pygeoapi#512]
**Description**
[GeoHealthCheck demo](https://demo.geohealthcheck.org/resource/165?lang=en) is finding an `id` issue with the GeoJSON "Lakes" dataset on [pygeoapi demo master](https://demo.pygeoapi.io/master/collections/ogr_geojson_lakes/items) when this dataset is served via the OGR Provider. See e.g. https://demo.pygeoapi.io/master/collections/ogr_geojson_lakes/items . All `get item` requests fail, e.g.: https://demo.pygeoapi.io/master/collections/ogr_geojson_lakes/items/1 . This was not the case before the `id` field was moved. 

**Steps to Reproduce**
* go to https://demo.pygeoapi.io/master/collections/ogr_geojson_lakes/items
* select an item
* see error response

**Expected behavior**
Item should be returned.

**Screenshots/Tracebacks**
The error comes from the Python OGR driver, the log:
```
[2020-08-11T09:27:28Z] {/pygeoapi/pygeoapi/provider/ogr.py:808} ERROR - Error Number: 1, Type: Failure, Msg: "id" not recognised as an available field.
[2020-08-11T09:27:28Z] {/pygeoapi/pygeoapi/provider/ogr.py:397} ERROR - <built-in function Layer_SetAttributeFilter> returned a result with an error set
```

**Environment**
 - OS: any
 - Python version: 3
 - pygeoapi version: latest master as on aug 11, 2020

**Additional context**
This error was found after #499 was solved. The `id` field was moved to the `Feature` level.
Apparantly OGR does not like this, or it may be a setting.

**Comment by justb4** [comment]
Possibly for GeoJSON [GetFeature](https://gdal.org/python/osgeo.ogr.Layer-class.html#GetFeature) may need to be called, i.s.o. `SetAttribute` and [GetNextFeature](https://gdal.org/python/osgeo.ogr.Layer-class.html#GetNextFeature), and `id-field` ignored.

**Comment by tomkralidis** [comment]
@justb4 FYI we merged #515 (RE: #499).  Does this issue still occur given current master now?

**Comment by justb4** [comment]
@tomkralidis @alex-mathew  this issue is solved, but the [GHC Demo Probe still fails](https://demo.geohealthcheck.org/resource/165?lang=en), but now on the Lakes dataset served via the GeoJSON Provider, as getting an Item by Id fails (The GHC Probe crawls all Collections, first getting multiple items, limit 1, and then for that Item
 attempting to fetch by its `id`, e.g. https://demo.pygeoapi.io/master/collections/lakes/items/1, which fails.
May be a minor mishap in the GHC Probe, as strictly Feature Id (at Feature level) is not required for  GeoJSON Features. Not sure what OAPIF standard requires.

But IMHO a Feature Id at Feature Level should be provided by the Lakes dataset - source [RFC 7946](https://tools.ietf.org/html/rfc7946):

*If a Feature has a commonly used identifier, that identifier SHOULD be included as a member of the Feature object with the name "id", and the value of this member is either a JSON string or number.*

The broader issue is how `pygeoapi` and possibly also the OAPIF standard should deal with "Feature Id" for GeoJSON, respecting the GeoJSON IETF Standard. We can close this one.  I will open a new issue, where we first should clarify what the standards require and how to comply within `pygeoapi`.

**Comment by alex-mathew** [comment]
@justb4 @tomkralidis  Actually I just noticed that there was an issue with the fix I did. The get method was failing because of the type mismatch of the identifier. The ids in the geojson file were integers and the id fetched from the URL will be a string. The pytest contained ids of type string. That's why it didn't fail.
